### ISSUE TITLE
Guide / Wiki / Documentation

### ISSUE THREAD

**Issue by meganthomas** [RazielXT/mtTorrent#8]
Hi RazielXT, 

The project is very good. Is it possible to get some write about the code. Current all the code is missing comments and is bit hard to follow. For starter like me if some level of comments in the GUI client that would be very helpful and then secondly with the mttorrent project. Best would be if you get a wiki or detailed documentation. 

Warm regards
Megan

**Comment by RazielXT** [comment]
Hi!

I agree that GUI code is messy as it wasnt primary objective of the project. I considered rewriting it to C# to make it more modern. For now I will try to refactor the existing code to be more readable. For easiest example please check SdkExample project.

I will look into creating some readable documentation about usage, and add comments to API interface.

**Comment by meganthomas** [comment]
Thank you RazielXT. 

Kindly not that this might just be for me, since I dont have a public IP, SDK project doesnt work for me. Its does not download the file at all and gets stuck where as the GUI project works. Initially I thought it was because of DHT configuration in the API but even after modifying it, it still doesn't work.

**Comment by RazielXT** [comment]
You are right, DHT should be enabled before magnet link download. I refactored API a bit + added API call comments + extended SdkExample to improve reporting current torrent state.

**Comment by RazielXT** [comment]
I have refactored GUI code into hopefully something more readable. I have tested most basic scenarios but its not fully tested yet.
